string text
string mode
---
int8 result